base/reduce.jl

```diff
@@ -281,15 +281,29 @@ mapreduce_impl(f, op, A::AbstractArrayOrBroadcasted, ifirst::Integer, ilast::Int
     mapreduce(f, op, itrs...; [init])
 
 Apply function `f` to each element(s) in `itrs`, and then reduce the result using the binary
-function `op`. If provided, `init` must be a neutral element for `op` that will be returned
-for empty collections. It is unspecified whether `init` is used for non-empty collections.
-In general, it will be necessary to provide `init` to work with empty collections.
+function `op`.
+
+The order of function evaluations and the associativity of the reduction is unspecified. Some
+implementations may reuse the return value of `f` for elements that appear multiple times in
+`itr`. Use [`mapfoldl`](@ref) or [`mapfoldr`](@ref) for
+strict left or right associativity and guaranteed invocation of `f` for every value.
+
+If provided, `init` serves as the return value for empty `itrs`. For non-empty iterators,
+it is included in the reduction exactly once and ensures that every element in `itrs` is
+used as an argument to `op`. Like the reduction itself, the exact order and associativity
+of how `init` is included is not specified. It is generally an error to call `mapreduce`
+with empty collections without specifying an `init` value, but in unambiguous cases the
+identity value may be returned; see [`Base.reduce_empty`](@ref) for more details.
 
 [`mapreduce`](@ref) is functionally equivalent to calling
-`reduce(op, map(f, itr); init=init)`, but will in general execute faster since no
+`reduce(op, map(f, itrs...)...; init=init)`, but will in general execute faster since no
```

Comment:
```suggestion
`reduce(op, map(f, itrs...); init=init)`, but will in general execute faster since no
```
This got a little carried away with the dots here.